llvm/test/MC/Sparc/sparc-asm-errors.s

```diff
@@ -11,7 +11,7 @@
         ! V9: unknown membar tag
         membar #BadTag
 
-        ! V8: instruction requires a CPU feature not currently enabled
+        ! V8: unexpected token
         ! V9: invalid membar mask number
         membar -127
```

Comment:
There are many places where ParseStatus is not handled correctly.
In this case, it is
```
    if (!parseOperand(Operands, Name).isSuccess()) {
      SMLoc Loc = getLexer().getLoc();
      return Error(Loc, "unexpected token");
    }
```
It should propagate the error returned by parseOperand instead of issuing another one.
Something like:
```
    ParseStatus R = parseOperand(Operands, Name).isSuccess();
    if (R.isFailure())
      return true;
    if (R.isNoMatch())
      return TokError( "unexpected token");
    // success, proceed do the next operand
```

Reply:
It appears changing this line reveals other bugs. It may be worth making this change separately, before enabling ParseForAllFeatures.